### PR TITLE
 #184842234 Automated Postgresql Backups with NodeJS

### PR DESCRIPTION
**Pull request by Martha-Iradukunda**
#### What does this PR do?
* This PR adds the feature of an automated backups for PostgreSQL databases using NodeJS. This feature helps to ensure data safety and recoverability in case of failures or disasters.

#### Description of Task to be completed?
* The task is to research and select a suitable NodeJS package or library for PostgreSQL backups, such as pg_dump or node-postgres, set up a backup configuration file with backup schedule, retention policy, and target storage options, implement a backup script in NodeJS with error handling and logging mechanisms, test the backup script in a staging environment with different database sizes and configurations, set up automated backup job scheduling using a task runner or cron job, monitor backup performance and storage usage, and optimize as needed. Additionally, a restore script must be implemented in NodeJS with point-in-time recovery options and validation checks and tested in a staging environment with different backup sources and configurations. Finally, the backup and restore process must be documented with instructions for configuration, testing, and recovery.
#### How should this be manually tested?
* The backup and restore process should be manually tested in a staging environment with different database sizes and configurations to ensure that it is automated, reliable, and scalable with minimal impact on database performance. Backup performance and storage usage should also be monitored and optimized as needed.
#### Any background context you want to provide?
*  Automated backups are crucial for ensuring data safety and recoverability in case of failures or disasters. Implementing an automated backup system can help database administrators minimize the risk of data loss and avoid downtime. NodeJS is a popular programming language for building scalable and reliable server-side applications, making it an ideal choice for creating a backup system for PostgreSQL databases. Adding this feature to an existing e-commerce web application can help ensure the integrity and availability of the application's data.

#### What are the relevant pivotal tracker/Trello stories?
* [#184842234](https://www.pivotaltracker.com/n/projects/2632830/stories/184842234)

#### Screenshots (if appropriate)
* Screen Shot 2023-04-06 at 01 28 31
* Screen Shot 2023-04-06 at 01 34 17
* Screen Shot 2023-04-06 at 01 34 46

#### Questions:
N/A

### CODE REVIEW

File: src/pgbackup/backup.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+const backup = () => {
+  const date = moment().format('YYYYMMDD-HHmmss');
+  const filename = `backup-${date}.sql`;
+
+  const cmd = `pg_dump -h ${config.db.host} -p ${config.db.port} -U ${config.db.user} -F p -b -v -f ${filename} ${config.db.database}`;
+
+  exec(cmd, (error, stdout, stderr) => {
+    if (error) {
+      console.error(`Backup failed: ${error.message}`);
```
Comment (nshutijonathan) [inline src/pgbackup/backup.js:19]: return not console

File: src/pgbackup/backup.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    } else {
+      console.log(`Backup successful: ${filename}`);
```
Comment (nshutijonathan) [inline src/pgbackup/backup.js]: return not console

File: src/pgbackup/backup.restore.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/* eslint-disable no-unused-vars */
+import { exec } from 'child_process';
+import config from './backupconfig';
+
+const restore = (filename) => {
+  const cmd = `pg_restore -h ${config.db.host} -p ${config.db.port} -U ${config.db.user} -d ${config.db.database} ${filename}`;
+
+  exec(cmd, (error, stdout, stderr) => {
+    if (error) {
+      console.error(`Restore failed: ${error.message}`);
```
Comment (nshutijonathan) [inline src/pgbackup/backup.restore.js]: return not console

File: src/pgbackup/backup.restore.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    } else {
+      console.log(`Restore successful: ${filename}`);
```
Comment (nshutijonathan) [inline src/pgbackup/backup.restore.js]: return not console

**Review by nshutijonathan — COMMENTED** [review]
remove console.log

**Review by Habihirwe — APPROVED** [review]
good job

**Review by edwinnambaje — APPROVED** [review]
LGTM

**Review by INValentin — APPROVED** [review]
LGTM

**Review by nshutijonathan — APPROVED** [review]
LGTM

**Pull request merged**